radicle-httpd/src/api/v1/projects.rs

```diff
@@ -131,7 +131,7 @@ async fn history_handler(
     };
 
     let storage = &ctx.profile.storage;
-    let repo = storage.repository(project)?;
+    let repo = Repository::open(storage.path())?;
```

Comment:
It is confusing that we have two `Repository` types, one from `storage` and one from` radicle-surf`. Hopefully we can merge them somehow.

Reply:
`storage.path()` does not point to a repository though. I guess you'd have to do:
`let repo = Repository::from(repo.raw())` if you had a `From<git2::Repository>` instance.

Reply:
I see. `storage` can have multiple resources, where each resource is a git repo, and `storage.path()` is the parent directory of the actual git repo. Is that the case? I will check what's the proper way to open the repo here.

Reply:
correct!

Reply:
Ya, it's probably best to do the `From<git2::Repository>`. The other option which I mentioned waaaaay back was to design surf using traits :) But that would be a lot more work now.

Reply:
I've updated the diff to grab the correct (hopefully) repo path based on the `storage` and the `project`. 

I'm trying to stick with getting the path string because: "Write programs to handle text streams, because that is a universal interface." (UNIX).  ;-)

Reply:
Btw, what is our strategy for writing test for `radicle-httpd`?

Reply:
We have tests that we need to port from client-services, see here for eg.: https://github.com/radicle-dev/radicle-client-services/blob/master/http-api/src/v1/projects.rs#L977